onnx/defs/math/defs.cc

```diff
@@ -3392,4 +3392,591 @@ ONNX_OPERATOR_SET_SCHEMA(
+                AttributeProto::INT,
+                static_cast<int64_t>(0))
+            .Input(
+                0,
+                "input",
+                "For real input, the following shape is expected: [batch_idx][n_fft]."
+                "For complex input, the following shape is expected: [batch_idx][n_fft][2]." 
+                "The final dimension represents the real and imaginary parts of the value."
+                "For real multi-dimensional input, the following shape is expected: [batch_idx][signal_dim1][signal_dim2]...[signal_dimN][1]."
+                "For complex multi-dimensional input, the following shape is expected: [batch_idx][signal_dim1][signal_dim2]...[signal_dimN][2]."
+                "The first dimension is the batch dimension.",
+                "T",
+                OpSchema::Single,
+                true,
+                1,
+                OpSchema::NonDifferentiable)
+            .Output(
+                0,
+                "output",
+                "The Fourier Transform of the input vector."
+                "If signal_dimN = 1, and onesided is 0, [batch_idx][n_fft][2]"
+                "If signal_dimN = 1, and onesided is 1, [batch_idx][floor(n_fft/2)+1][2]" 
+                "If signal_dimN = 2, and onesided is 0 and axis = 0, [batch_idx][signal_dim1][signal_dim2][2]"
+                "If signal_dimN = 2, and onesided is 0 and axis = 1, [batch_idx][signal_dim1][signal_dim2][2]"
+                "If signal_dimN = 2, and onesided is 1 and axis = 0, [batch_idx][floor(signal_dim1/2)+1][signal_dim2][2]"
+                "If signal_dimN = 2, and onesided is 1 and axis = 1, [batch_idx][signal_dim1][floor(signal_dim2/2)+1][2]",
+                "T")
+            .TypeConstraint(
+                "T",
+                {"tensor(float16)", "tensor(float)", "tensor(double)", "tensor(bfloat16)"},
+                "Constrain input and output types to float tensors.")
+            .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+                propagateElemTypeFromInputToOutput(ctx, 0, 0);
+                const int64_t batch_ndim = 1;
+
+                auto& input_shape = getInputShape(ctx, 0);
+                auto dim_size = static_cast<int64_t>(input_shape.dim_size());
+                auto has_component_dimension = dim_size > 2; 
+
+                ONNX_NAMESPACE::TensorShapeProto result_shape_proto = input_shape;
+                
+                bool axis = static_cast<bool>(getAttribute(ctx, "axis", 0));
+                bool is_onesided = static_cast<bool>(getAttribute(ctx, "onesided", 0));
+                if (is_onesided) {
+                    // Since signal_ndim = 1, and multidimensional DFT is not supported,
+                    // only the single signal dim (1) needs to be updated
+                    auto n_fft = input_shape.dim(1 + axis).dim_value();
```

Comment:
Here, the specific `dim_value` may not be known.

Reply:
added check.